infrahub_sdk/node.py

```diff
@@ -1479,15 +1479,15 @@ async def _process_relationships(
         for rel_name in self._relationships:
             rel = getattr(self, rel_name)
             if rel and isinstance(rel, RelatedNode):
-                relation = node_data["node"].get(rel_name)
-                if relation.get("node", None):
+                relation = node_data["node"].get(rel_name, None)
+                if relation and relation.get("node", None):
```

Comment:
This is the only change that's required for the other ones `relation = node_data["node"].get(rel_name)` would already return `None` so we don't need a default value there.